dist/angular-oauth2.js

```diff
@@ -69,37 +47,40 @@
         revokePath: "/oauth2/revoke"
     };
     var requiredKeys = [ "baseUrl", "clientId", "grantPath", "revokePath" ];
+    function cleanConfig(config) {
+        angular.forEach(requiredKeys, function(key) {
+            if (!config[key]) {
+                throw new Error("Missing parameter: " + key + ".");
+            }
+        });
+        if ("/" === config.baseUrl.substr(-1)) {
+            config.baseUrl = config.baseUrl.slice(0, -1);
+        }
+        if ("/" !== config.grantPath[0]) {
+            config.grantPath = "/" + config.grantPath;
+        }
+        if ("/" !== config.revokePath[0]) {
+            config.revokePath = "/" + config.revokePath;
+        }
+        return config;
+    }
     function OAuthProvider() {
-        var config;
+        var defaultConfig;
         this.configure = function(params) {
-            if (config) {
+            if (defaultConfig) {
                 throw new Error("Already configured.");
             }
             if (!(params instanceof Object)) {
                 throw new TypeError("Invalid argument: `config` must be an `Object`.");
             }
-            config = angular.extend({}, defaults, params);
```

Comment:
Instead of doing this, I think i'll be simpler to add a new argument to each method called `config`. Then on each method, you just need to extend `defaultConfig` with the those values.